blockauth/apple/throttles.py

```
@@ -0,0 +1,93 @@
+    default_rate: tuple[int, int] = (30, 60)
+
+    def __init__(self):
+        self.num_requests, self.duration = _rate_tuple(self.setting_name, self.default_rate)
+        self.history: list[float] = []
+        self.now = self.timer()
+
+    def get_cache_key(self, request) -> str:
+        ip = get_client_ip(request) or "unknown"
+        return f"apple_throttle_{self.scope}_{ip}"
+
+    def allow_request(self, request, view) -> bool:
```

Ruff (0.15.12)

[warning] Unused method argument: `view`

(ARG002)